scrapy_autounit/utils.py

```diff
@@ -172,276 +43,12 @@ def get_spider_class(spider_name, project_settings):
-            print("Testing fixture '%s'" % (f))
-            test = generate_test(os.path.abspath(file_path))
-            test(self)
-
-
-if __name__ == '__main__':
-    unittest.main()
-'''.format(
-        test_name=test_name,
-        command=command,
-        url=url,
-    )
-
-    with open(str(test_path), 'w') as f:
-        f.write(test_code)
-
-
-def binary_check(fx_obj, cb_obj, encoding):
-    if isinstance(cb_obj, (dict, Item)):
-        fx_obj = {
-            key: binary_check(value, cb_obj[key], encoding)
-            for key, value in fx_obj.items()
-        }
-
-    if isinstance(cb_obj, list):
-        fx_obj = [
-            binary_check(fxitem, cbitem, encoding)
-            for fxitem, cbitem in zip(fx_obj, cb_obj)
-        ]
-
-    if isinstance(cb_obj, Request):
-        headers = {}
-        for key, value in fx_obj['headers'].items():
-            key = to_bytes(key, encoding)
-            headers[key] = [to_bytes(v, encoding) for v in value]
-        fx_obj['headers'] = headers
-        fx_obj['body'] = to_bytes(fx_obj['body'], encoding)
-
-    if isinstance(cb_obj, six.binary_type):
-        fx_obj = fx_obj.encode(encoding)
-
-    return fx_obj
-
-
-def set_spider_attrs(spider, _args):
-    for k, v in _args.items():
-        setattr(spider, k, v)
-
-
-def parse_callback_result(result, spider):
-    processed_result = []
-    out = []
-    for elem in result:
-        out.append(elem)
-        is_request = isinstance(elem, Request)
-        if is_request:
-            _data = parse_request(elem, spider)
-        else:
-            _data = parse_object(copy.deepcopy(elem), spider)
-        processed_result.append({
-            'type': 'request' if is_request else 'item',
-            'data': _data
-        })
-    return processed_result, out
-
-
-def prepare_callback_replay(fixture_path, encoding="utf-8"):
-    with open(str(fixture_path), 'rb') as f:
-        raw_data = f.read()
-
-    fixture_info = unpickle_data(decompress_data(raw_data), encoding)
-    if 'fixture_version' in fixture_info:
-        encoding = fixture_info['encoding']
-        data = unpickle_data(fixture_info['data'], encoding)
-    else:
-        data = fixture_info  # legacy tests
-
-    settings = get_project_settings()
-
-    spider_name = data.get('spider_name')
-    if not spider_name:  # legacy tests
-        spider_name = os.path.basename(
-            os.path.dirname(
-                os.path.dirname(fixture_path)
-            )
-        )
-
-    spider_cls = get_spider_class(spider_name, settings)
-    spider_cls.update_settings(settings)
-
-    for k, v in data.get('settings', {}).items():
-        settings.set(k, v, 50)
-
-    crawler = Crawler(spider_cls, settings)
-    spider_args_in = data.get('spider_args', data.get('spider_args_in', {}))
-    spider = spider_cls.from_crawler(crawler, **spider_args_in)
-    crawler.spider = spider
-
-    return data, crawler, spider, settings
+def python_version():
+    return 2 if six.PY2 else 3
```

Comment:
I think it might not be worthy to have `six` as a dependency only for this usage. `sys.version_info.major` should do the trick.

Reply:
That's true. I guess we used six for other things before.

Reply:
Solved this here: https://github.com/scrapinghub/scrapy-autounit/pull/81/commits/58c3534ad9d25bbe5057ef23bc712e537619ab1a